### ISSUE TITLE
Inconsistent output file path when bundling with `--compile`

### ISSUE THREAD

**Issue by jkbz64** [oven-sh/bun#8587]
### What version of Bun is running?

1.0.25+a8ff7be64

### What platform is your computer?

Linux 6.6.3-411.asahi.fc39.aarch64+16k aarch64 unknown

### What steps can reproduce the bug?

1. Create this tree and run `bun add @napi-rs/canvas`
```
.
├── a
│   └── index.js
├── b
│   └── index.js
├── bun.lockb
└── package.json
```
a/index.js contains just `console.log('whatever')`
b/index.js contains `import "@napi-rs/canvas"`

2. Run in the root directory:
`bun build --target bun --compile a/index.js --outfile ./a-out` 
`bun build --target bun --compile b/index.js --outfile ./b-out`
4. Run `tree`

`b-out` is created inside b directory, it should be in the root directory

### What is the expected behavior?

```
.
├── a
│   └── index.js
├── a-out
├── b
│   └── index.js
├── b-out
├── bun.lockb
└── package.json
```

### What do you see instead?

```
.
├── a
│   └── index.js
├── a-out
├── b
│   ├── b-out
│   └── index.js
├── bun.lockb
└── package.json
```

### Additional information

It might be napi specific.

**Comment by pablodelamerced** [comment]
Hello everyone,

I hope you're well. I've encountered a specific issue when using the @napi-rs/canvas or canvas libraries in our project. I wanted to share this here to seek a solution or to see if it's already being addressed.

## Issue Description:
When integrating the @napi-rs/canvas or canvas library and compiling the project using the command bun build --compile ./src/index.ts --outfile=binary, the compiled file does not end up in the project's root directory as expected, but instead within the src/ folder. This unexpected location of the compiled file could cause confusion and complications in the deployment and execution processes of the project.

## Steps to Reproduce:

- Install the @napi-rs/canvas or canvas library into the project.
- Compile the project with the command bun build --compile ./src/index.ts --outfile=binary.
- Observe the location of the compiled output file.


The compiled files are supposed to be located in the project's root directory. The compiled file is found in the src/ folder:

<img width="974" alt="Captura de pantalla 2024-02-05 a las 17 43 53" src="https://github.com/oven-sh/bun/assets/65915485/dc6ae918-a4cc-4fcc-bab0-bf8e5c6b6576">

Thanks!

**Comment by jkbz64** [comment]
Upgrading to v1.1.5 fixed this issue for me (probably thanks to #10385)
